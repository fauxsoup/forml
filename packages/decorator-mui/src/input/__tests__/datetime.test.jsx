import DateTime from '../datetime';
import Context from '@forml/context';
import React from 'react';
import { render } from '@testing-library/react';
import * as decorator from '../../';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import moment from 'moment';
import 'moment-timezone';

moment.tz.setDefault('GMT');

describe('renders', function () {
    let form;
    let inputValue;

    beforeEach(function () {
        form = { type: 'datetime' };
        inputValue = new Date(0).toISOString();
    });

    describe('with form options', function () {
        let fields = {
            disablePast: [true, false],
            disableFuture: [true, false],
            variant: ['inline', 'dialog'],
            fullWidth: [true, false],
            autoOk: [true, false],
            openTo: ['datetime', 'year', 'month'],
            format: ['YYYY/MM/DD HH:mm:ss', 'LLLL'],
            readonly: [true, false],
        };

        Object.keys(fields).forEach(function (field) {
            fields[field].forEach(function (value) {
                describe(`${field}`, function () {
                    test(`${value}`, function () {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <LocalizationProvider dateAdapter={AdapterMoment}>
                                <Context.Provider value={{ decorator }}>
                                    <DateTime form={form} value={inputValue} />
                                </Context.Provider>
                            </LocalizationProvider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
