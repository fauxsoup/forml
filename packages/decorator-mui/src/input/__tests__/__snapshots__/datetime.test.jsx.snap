// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with form options autoOk false 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options autoOk true 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options disableFuture false 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options disableFuture true 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options disablePast false 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options disablePast true 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="true"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options format LLLL 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options format YYYY/MM/DD HH:mm:ss 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options fullWidth false 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options fullWidth true 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options openTo datetime 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options openTo month 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options openTo year 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options readonly false 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options readonly true 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input Mui-disabled css-1x51dt5-MuiInputBase-input-MuiInput-input"
      disabled=""
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options variant dialog 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;

exports[`renders with form options variant inline 1`] = `
<div>
  <div
    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-1tcs2i1-MuiInputBase-root-MuiInput-root"
  >
    <input
      aria-invalid="false"
      aria-label="Choose date, selected date is Jan 1, 1970"
      aria-readonly="true"
      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      readonly=""
      type="text"
      value="01/01/1970 12:00 AM"
    />
  </div>
</div>
`;
